modules/bls-dkg/test/bls-dkg.js

```diff
@@ -8,113 +8,139 @@ const {
+  });
 
-  let walletKey
+  let walletKey;
 
   before('combines shares into wallet signing key', function() {
-    walletKey = mergeSecretShares([userSecretShares[2], backupSecretShares[2], walletSecretShares[2]])
-  })
+    walletKey = mergeSecretShares([userSecretShares[2], backupSecretShares[2], walletSecretShares[2]]);
+  });
 
-  let userSig
+  let userSig;
 
   before('user signs message', async function() {
-    userSig = await sign(message, userKey)
-  })
+    userSig = await sign(message, userKey);
+  });
 
-  let backupSig
+  let backupSig;
 
   before('backup signs message', async function() {
-    backupSig = await sign(message, backupKey)
-  })
+    backupSig = await sign(message, backupKey);
+  });
 
-  let walletSig
+  let walletSig;
 
   before('wallet signs message', async function() {
-    walletSig = await sign(message, walletKey)
-  })
+    walletSig = await sign(message, walletKey);
+  });
 
   it('verifies user-backup sig', async function() {
     const userBackupSig = mergeSignatures({
       1: userSig,
       2: backupSig,
-    })
+    });
     if (!await verify(userBackupSig, message, commonPub)) {
-      throw new Error('Could not verify user + backup signature')
+      throw new Error('Could not verify user + backup signature');
     }
-  })
+  });
 
   it('verifies user-wallet sig', async function() {
     const userWalletSig = mergeSignatures({
       1: userSig,
       3: walletSig,
-    })
+    });
     if (!await verify(userWalletSig, message, commonPub)) {
-      throw new Error('Could not verify user + wallet signature')
+      throw new Error('Could not verify user + wallet signature');
     }
-  })
+  });
 
   it('verifies backup-wallet sig', async function() {
     const backupWalletSig = mergeSignatures({
       2: backupSig,
       3: walletSig,
-    })
+    });
     if (!await verify(backupWalletSig, message, commonPub)) {
-      throw new Error('Could not verify backup + wallet signature')
+      throw new Error('Could not verify backup + wallet signature');
+    }
+  });
+
+  it('verifies user-backup subkey sig with public derived', async function() {
+    const userPublicShareDKG = publicShare(userPolynomial);
+    const derivedUserPublicShareDKG = publicDerive(userPublicShareDKG, subKeyPath, userPolynomial[1]);
+    if (derivedUserPublicShareDKG.pk !== userPublicShare) {
+      throw new Error('Derived public key is incorrect');
+    }
+    if (derivedUserPublicShareDKG.chaincode !== userSubKey.chaincode) {
+      throw new Error('Derived public chaincode is incorrect');
+    }
+    const pub = mergePublicShares([derivedUserPublicShareDKG.pk, backupPublicShare, walletPublicShare]);
+    const userBackupSig = mergeSignatures({
+      1: userSig,
+      2: backupSig,
+    });
+
+    if (!await verify(userBackupSig, message, pub)) {
+      throw new Error('Could not verify user + backup signature');
     }
-  })
+  });
 
```

Comment:
Please also add a test that verifies `mergePublicShares([publicDerive(pk_A, chaincode_Common, path), pk_B, pk_C]) == publicDerive(mergePublicShares([pk_A, pk_B, pk_C]), chaincode_Common, path)`

Reply:
This is interesting, I can't make it work. Doesn't match between them.

Reply:
My mistake, this wouldn't work because the offset for each subkey is going to be different between the user key and the common key in the derivation path. Instead you need to rewrite the `verifies user-backup subkey sig with public derived` test;
1) The user derives a new secret share `derivedUserSk`.
2) Multiply `derivedUserSk` by the curve's base to get `derivedUserPublicShare`.
3) When you merge the derived user public share with the other public shares, you get a new `derivedCommonPub`.
4) `derivedCommonPub1` should be equal to `derivedCommonPub2 = publicDerive(commonPub, commonChaincode, path)`
5) Signature sholud verify against the newly derived common public key.

Reply:
Makes sense, will do that UT.